compiler/src/dotty/tools/dotc/core/NullOpsDecorator.scala

```diff
@@ -50,6 +50,18 @@ object NullOpsDecorator:
       val stripped = self.stripNull()
       stripped ne self
     }
+
+    def admitsNull(using Context): Boolean = {
```

Comment:
Since we are basically checking if a type has `Null` as a subtype, I'm not sure whether we should widen a singleton type. 

For example, 

```scala
val s: String | Null = ???
val ss: s.type | Null = ???
val snotnull: s.type = ss.nn
```

I'd like to allow `ss.nn` to say if it returns a value, the value is equal to `s` and not null.